relay-server/src/actors/envelopes.rs

```diff
@@ -1681,41 +1686,53 @@ impl EnvelopeProcessor {
     }
 
     /// Extract metrics for transaction events with breakdowns and measurements.
-    #[cfg(feature = "processing")]
     fn extract_transaction_metrics(
         &self,
         state: &mut ProcessEnvelopeState,
     ) -> Result<(), ProcessingError> {
-        let config = match state.project_state.config.transaction_metrics {
+        let config = match state.project_state.config().transaction_metrics {
             Some(ErrorBoundary::Ok(ref config)) => config,
             _ => return Ok(()),
         };
 
-        let breakdowns_config = state.project_state.config.breakdowns_v2.as_ref();
-        let conditional_tagging_config = state
-            .project_state
-            .config
-            .metric_conditional_tagging
-            .as_slice();
+        if !config.is_enabled() {
+            return Ok(());
+        }
 
-        if let Some(event) = state.event.value() {
-            let extracted_anything;
+        if let Some(item) = state.transaction_item.as_mut() {
+            if item.metrics_extracted() {
+                return Ok(());
+            }
 
-            metric!(
-                timer(RelayTimers::TransactionMetricsExtraction),
-                extracted_anything = &extracted_anything.to_string(),
-                {
-                    // Actual logic outsourced for unit tests
-                    extracted_anything = extract_transaction_metrics(
-                        config,
-                        breakdowns_config,
-                        conditional_tagging_config,
-                        event,
-                        &mut state.extracted_metrics,
-                    );
-                }
-            );
-            Ok(())
+            let breakdowns_config = state.project_state.config.breakdowns_v2.as_ref();
+            let conditional_tagging_config = state
+                .project_state
+                .config
+                .metric_conditional_tagging
+                .as_slice();
+
+            let (event, _) = self.event_from_json_payload(item, Some(EventType::Transaction))?;
```

Comment:
I don't think this should be necessary. Don't you have access to `state.event` here?